posthog/hogql_queries/query_runner.py

```diff
@@ -57,6 +58,15 @@
 DataT = TypeVar("DataT")
 
 
+class RecalculationMode(Enum):
+    REQUEST = auto()
+    """Always recalculate, except if a VERY recent result is present in cache."""
+    IF_STALE = auto()
+    """Use cache, unless the results are stale or missing."""
+    NEVER = auto()
+    """Only use cache."""
```

Comment:
Let's be absolutely clear:

```suggestion
    ALWAYS_EXCEPT_RECENT = auto()
    """Always recalculate, except if a VERY recent result is present in cache."""
    USE_CACHE_UNLESS_STALE = auto()
    """Use cache, unless the results are stale or missing."""
    ONLY_USE_CACHE = auto()
    """Only use cache."""
```

Reply:
Oh yeah, this _was_ pretty vague. Renamed somewhat along the lines of your suggestion!